metaci/build/autoscaling.py

```diff
@@ -99,29 +100,39 @@ class HerokuAutoscaler(Autoscaler):
     API_ROOT = "https://api.heroku.com/apps"
 
     def scale(self):
-        worker_type = settings.WORKER_DYNO_NAME
-        url = f"{self.API_ROOT}/{settings.HEROKU_APP_NAME}/formation/{worker_type}"
-        headers = {
+        self.worker_type = settings.WORKER_DYNO_NAME
+        self.url = (
+            f"{self.API_ROOT}/{settings.HEROKU_APP_NAME}/formation/{self.worker_type}"
+        )
+        self.headers = {
             "Accept": "application/vnd.heroku+json; version=3",
             "Authorization": f"Bearer {settings.HEROKU_TOKEN}",
         }
```

Comment:
Since you're turning worker_type, url, and headers into instance attributes, it would make more sense to set them in __init__. That avoids the potential for something to try to use them when they haven't been set up yet.